Docs: create conda env for building docs (#3074)
ComputationalRadiationPhysics/picongpu: docs/source/dev/sphinx.rst

    # python tools & style theme
    pip install -r requirements.txt # --user

> However, in order to not break any of your existing Python configurations you can also create a new environment that you only use for building the documentation.

Reviewer: Required: there needs to be a comma in `... configurations, you can ...`
Optional: remove `However,`

    # python tools & style theme
    pip install -r requirements.txt # --user

In order to not break any of your existing Python configurations, you can also create a new environment that you only use for building the documentation.

> Since it is possible to install doxygen with conda, the following method employs this way.

Reviewer 2: ```suggestion
Since it is possible to install doxygen with conda, the following demonstrates this.
```

Author: Implemented your suggestions in one commit

.. code-block:: bash

    cd docs/

    # create a bare conda environment containing just all the requirements
    # for building the picongpu documentation
    # note: also installs doxygen inside this environment

>     conda env create picongpu-docs-env.yml

Reviewer: when I run thi scommand, I get the following error:
```bash
SpecNotFound: Invalid name, try the format: user/package
```

Reviewer: Might be darwin (MacOS) related issue.

Reviewer: I am using `conda 4.4.10`

Author: Haha, sorry, the `--file` was missing which is correct in the file comment itself 
Good find!

>     conda env create picongpu-docs-env.yml

Reviewer: ```suggestion
    conda env create --file picongpu-docs-env.yml
```

>     # start up the environment

Reviewer: ```suggestion
    # start up the environment as suggested during its creation e.g.
```

>     conda activate picongpu-docs-env

Reviewer: ```suggestion
    conda activate picongpu-docs-env
    # or
    source activate picongpu-docs-env    
```
